poh-bundle-linux-x64/oh.sh

```diff
@@ -1,68 +1,197 @@
+echo ""
+
+######## Environment check:
+
+######## OPENHOSPITAL Configuration
+# POH_PATH is the directory where Portable OpenHospital files are located
+# POH_PATH=/usr/local/OpenHospital
+
+# name of this shell script
+SCRIPT_NAME="oh.sh"
+
+# set current dir
+CURRENT_DIR=$PWD
+
+if [ -z $POH_PATH ]; then
+	echo "Warning - POH_PATH not found. Please set it up properly."
+		if [ ! -f ./$SCRIPT_NAME ]; then
+		echo "Error - oh.sh not found in the current PATH. Please cd the directory where POH was unzipped or set up POH_PATH properly."
+		exit 0
+		fi
+	POH_PATH=$PWD
+fi
+
+######## Software configuration - change at your own risk :-)
 
 MYSQL_DIR="mysql-5.7.30-linux-glibc2.12-x86_64"
-JAVA_DIR="jdk8u252-b09-jre"
+MYSQL_URL="https://downloads.mysql.com/archives/get/p/23/file"
+
+JAVA_DISTRO="OpenJDK11U-jdk_x64_linux_openj9_11.0.8_10_openj9-0.21.0"
+JAVA_URL="https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.8%2B10_openj9-0.21.0"
+JAVA_DIR="jdk-11.0.8+10"
+
+
 OH_DIR="oh"
+DATABASE_NAME="oh"
+DATABASE_USER="isf"
+DATABASE_PASSWORD="isf123"
 DICOM_DEFAULT_SIZE="4M"
 
-where_i_am=$(pwd)
-cd $(dirname $0)
-POH_PATH=$(pwd)
-cd $where_i_am
+cd $POH_PATH
 
-# Find a free port to run MySQL starting from the default port.
-mysql_port=3306
-while [ $(ss -tna | awk '{ print $4 }' | grep ":$mysql_port") ]; do
-	mysql_port=$(expr $mysql_port + 1)
-done
+######################## DO NOT EDIT BELOW THIS LINE ########################
+
+echo "Checking for software...."
+
+if [ ! -d "$POH_PATH/$MYSQL_DIR" ]; then
+
+		if [ ! -f "$POH_PATH/$MYSQL_DIR.tar.gz" ]; then
+
+		echo "Warning - MySQL  not found. Do you want to download it ? (-630 Mb)"
+
+		read -p "(y/n)?" choice
+		case "$choice" in 
+			y|Y ) echo "yes";;
+			n|N ) echo "Exiting..."; exit 0;;
+			* ) echo "Invalid";;
+		esac
+
+		# Downloading mysql binary
+		echo "Downloading $MYSQL_DIR..."
+		wget $MYSQL_URL/$MYSQL_DIR.tar.gz
+		echo "Unpacking $MYSQL_DIR..."
+		tar zxvf $MYSQL_DIR.tar.gz
+		echo "Done !"
+	fi
+	echo "Using $MYSQL_DIR"
+fi
+
+
+if [ ! -d "$POH_PATH/$JAVA_DIR" ]; then
+
+	if [ ! -f "$POH_PATH/$JAVA_DISTRO.tar.gz" ]; then
+
+		echo "Warning - JAVA  not found. Do you want to download it ? (-180 Mb)"
+
+		read -p "(y/n)?" choice
+		case "$choice" in 
+			y|Y ) echo "yes";;
+			n|N ) echo "Exiting..."; exit 0;;
+			* ) echo "Invalid";;
+		esac
+
+		# Downloading openjdk binaries
+		echo "Downloading $JAVA_DISTRO..."
+		wget $JAVA_URL/$JAVA_DISTRO.tar.gz
+		echo "Unpacking $JAVA_DISTRO..."
+		tar zxvf $JAVA_DISTRO.tar.gz
+		echo "Done !"
+	fi
+fi
+
+######## Environment setup
+
+echo "Setting up environment...."
+
+# Find a free port to run MySQL starting from the default port
 
 POH_PATH_ESCAPED=$(echo $POH_PATH | sed -e 's/\//\\\//g')
+
+######## DICOM setup
+
 DICOM_MAX_SIZE=$(grep -i '^dicom.max.size' $POH_PATH/$OH_DIR/rsc/dicom.properties.ori  | cut -f2 -d'=')
 : ${DICOM_MAX_SIZE:=$DICOM_DEFAULT_SIZE}
 
+######## Database setup
+
+MYSQL_PORT=3307
```

Comment:
maybe let's move this to the top? It's easier to have all variables in the same place

Reply:
Agreed, done.